Content.Server/Gatherable/GatherableSystem.cs

```diff
@@ -60,29 +63,37 @@ private void OnDoAfter(EntityUid uid, GatherableComponent component, GatherableD
         if (args.Handled || args.Cancelled)
             return;
 
+        Gather(args.Args.Target.Value, component, tool.GatheringSound);
+        args.Handled = true;
+    }
+
+    public void Gather(EntityUid uid, GatherableComponent? component = null, SoundSpecifier? sound = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
         // Complete the gathering process
-        _destructible.DestroyEntity(args.Args.Target.Value);
-        _audio.PlayPvs(tool.GatheringSound, args.Args.Target.Value);
+        _destructible.DestroyEntity(uid);
+        _audio.PlayPvs(sound, uid);
 
         // Spawn the loot!
         if (component.MappedLoot == null)
             return;
 
-        var playerPos = Transform(args.Args.User).MapPosition;
+        var pos = Transform(uid).MapPosition;
 
         foreach (var (tag, table) in component.MappedLoot)
         {
             if (tag != "All")
             {
-                if (!_tagSystem.HasTag(tool.Owner, tag))
+                if (!_tagSystem.HasTag(uid, tag))
```

Comment:
Shouldn't this be a different uid passed into this method? uid is the equivalent of args.Args.Used before, not args.Args.Target.Value which is what this is now.